### PR TITLE
[#109] Added inventory UI 

### PR DESCRIPTION
**Pull request by Monkey-bored**
That was something.

### CODE REVIEW

File: development/code/Game.ts [hunk 1]
```diff
@@ -59,11 +60,16 @@ let deathCounter = 0;
...
 let isPieceKilled = false;
+let itemPlacedDownRecently = false;
 let fellOffTheBoardPiece: Piece | undefined;
 let isGameFinished = false;
 
 function initializeGame() {
   rulesManager = new RulesManager();
+
+  players.forEach((player) => {
+    initialiseInventoryUI(player.color);
```
Comment (Ido-Barnea) [inline development/code/Game.ts:71]: On land and sky, the British shall cry!!

File: development/code/Game.ts [hunk 2]
```diff
@@ -82,7 +88,13 @@ function endTurn() {
   }
 
   Logger.logMessages();
+
+  players.forEach((player) => {
+    changeShownInventory(player);
```
Comment (Ido-Barnea) [inline development/code/Game.ts:93]: `switchInventoryUI`?

File: development/code/Game.ts [hunk 3]
```diff
@@ -82,7 +88,13 @@ function endTurn() {
...
+  });
+
   renderScreen();
+  itemPlacedDownRecently = false;
```
Comment (Ido-Barnea) [inline development/code/Game.ts:97]: `wasItemPlacedThisTurn`?

File: development/code/LogicAdapter.ts [hunk 4]
```diff
@@ -56,7 +58,6 @@ export function onActionTriggered(
 
   const targetSquareId = getSquareIdByElement(targetElement);
   if (!targetSquareId) return;
-
```
Comment (Ido-Barnea) [inline development/code/LogicAdapter.ts:59]: Why did you remove the empty line...?

File: development/code/logic/items/Inventory.ts [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
 import { Item } from './Items';
 
 export class Inventory {
   items: Array<Item> = [];
+  player: Player;
 
-  addItem(item: Item, player: Player) {
+  constructor(_player: Player) {
+    this.player = _player;
+  }
```
Comment (Ido-Barnea) [inline development/code/logic/items/Inventory.ts:12]:
```
constructor(player: Player) {
	this.player = player;
}
```

File: development/code/ui/BoardManager.ts [hunk 6]
```diff
@@ -229,3 +229,6 @@ export function highlightLastMove(
   highlightSquare(targetSquareElement, true, false);
 }
 
+export function destroyItemOnInventory(itemElement: HTMLElement) {
```
Comment (Ido-Barnea) [inline development/code/ui/BoardManager.ts:232]: `destroyItemInInventory`

File: development/code/ui/Events.ts [hunk 7]
```diff
@@ -47,27 +47,27 @@ export function initializeEventListeners() {
...
 }
 
-export function dragPieceElement(element: HTMLElement) {
+export function dragElement(element: HTMLElement) {
   let startMouseX = 0;
   let startMouseY = 0;
   let endMouseX = 0;
   let endMouseY = 0;
-  element.onmousedown = dragPieceOnMouseDown;
+  element.onmousedown = dragOnMouseDown;
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:55]: `dragElementOnMouseDown`

File: development/code/ui/Events.ts [hunk 8]
```diff
@@ -76,41 +76,44 @@ export function dragPieceElement(element: HTMLElement) {
...
     document.onmousemove = null;
-
     const initialElement = draggedElement as HTMLElement;
     if (!initialElement) return;
 
-    let boardElement = initialElement.parentElement ?? undefined;
-    while (boardElement && !boardElement.classList.contains('board')) {
-      boardElement = boardElement.parentElement ?? undefined;
+    let parentContainer = initialElement.parentElement ?? undefined;
+    while (parentContainer && !(parentContainer.classList.contains('board') || parentContainer.classList.contains('player-inventory'))) {
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:90]:
```
while (
	parentContainer &&
	!(parentContainer.classList.contains('board') ||
	parentContainer.classList.contains('player-inventory'))
)
```

File: development/code/ui/Events.ts [hunk 9]
```diff
@@ -76,41 +76,44 @@ export function dragPieceElement(element: HTMLElement) {
...
+      parentContainer = parentContainer.parentElement ?? undefined;
     }
 
-    if (!boardElement) return;
-
+    if (!parentContainer) return;
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:94]: empty line after `if (!parentContainer) return;`

File: development/code/ui/Events.ts [hunk 10]
```diff
@@ -76,41 +76,44 @@ export function dragPieceElement(element: HTMLElement) {
...
 
-    if (droppedOnElement === undefined) {
-      triggerOnFellOffTheBoard(draggedElement, boardElement.id);
+
+    if (draggedElement.classList.contains('item')) {
+      if (!placeItemOnBoard(draggedElement, droppedOnElement)) {
+        returnItemToInventory(draggedElement);
+      }
+    }
+    else if (droppedOnElement === undefined) {
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:113]: To remain consistent, please change this to `{ else if (`

File: development/code/ui/Events.ts [hunk 11]
```diff
@@ -76,41 +76,44 @@ export function dragPieceElement(element: HTMLElement) {
...
         element.classList.contains('item') ||
-        element.classList.contains('piece') && element !== draggedElement;
+        element.classList.contains('piece') ) && element !== draggedElement;
     })[0];
 
-    if (droppedOnElement === undefined) {
-      triggerOnFellOffTheBoard(draggedElement, boardElement.id);
+
+    if (draggedElement.classList.contains('item')) {
+      if (!placeItemOnBoard(draggedElement, droppedOnElement)) {
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:109]: `if (x && y)` > `if () { if() }`

File: development/code/ui/InventoriesUI.ts [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  const inventoryItemElement = document.createElement('div') as HTMLElement;
+  inventoryItemElement.id = item.name;
+  inventoryItemElement.classList.add('item');
+  inventoryItemElement.classList.add('inventory-item');
+  inventoryItemElement.innerHTML = item.resource;
+  inventoryItemElement.draggable = true;
+  
+  
+
+  dragElement(inventoryItemElement);
```
Comment (Ido-Barnea) [inline development/code/ui/InventoriesUI.ts:31]: "No man's land"

File: development/code/ui/InventoriesUI.ts [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+  squareElement.setAttribute('player-color', playerColor);
+
+  playerInventoryElement.appendChild(squareElement);
+}
+
+export function switchShownInventory(playerColor: PlayerColors): boolean | undefined {
+  const playerInventoryElement = document.getElementById(playerColor);
+  if (!playerInventoryElement) {
+    return;
+  }
```
Comment (Ido-Barnea) [inline development/code/ui/InventoriesUI.ts:74]: Instead of returning `undefined`, why not return `false` (or `true`? What do you think is more correct?) if the `playerInventoryElement` doesn't exist?

**Review by Ido-Barnea — CHANGES_REQUESTED** [review]
See requested changes

File: development/code/Game.ts (hunk 1 above)
Comment (Monkey-bored) [reply development/code/Game.ts:71]: Ok and?

File: development/code/ui/Events.ts [hunk 14]
```diff
@@ -105,12 +105,9 @@ export function dragElement(element: HTMLElement) {
...
 
-    if (draggedElement.classList.contains('item')) {
-      if (!placeItemOnBoard(draggedElement, droppedOnElement)) {
-        returnItemToInventory(draggedElement);
-      }
-    }
-    else if (droppedOnElement === undefined) {
+    if (draggedElement.classList.contains('item') && !placeItemOnBoard(draggedElement, droppedOnElement)) {
+      returnItemToInventory(draggedElement);
+    } else if (droppedOnElement === undefined) {
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:110]: `else if (!droppedOnElement)` would be more general

**Review by Ido-Barnea — CHANGES_REQUESTED** [review]
One last change and we're good :)

File: development/code/LogicAdapter.ts (hunk 4 above)
Comment (Monkey-bored) [reply development/code/LogicAdapter.ts:59]: Mistakes were made

File: development/code/Game.ts [hunk 15]
```diff
@@ -216,12 +216,12 @@ function endGame() {
   isGameFinished = true;
 }
 
-function changeItemPlacedRecently() {
-  itemPlacedDownRecently = true;
+function changewasItemPlacedThisTurn() {
```
Comment (Ido-Barnea) [inline development/code/Game.ts]: `switchWasItemPlacedThisTurn`

File: development/code/Game.ts [hunk 16]
```diff
@@ -247,6 +247,6 @@ export const game = {
   setIsPieceKilled,
   getFellOffTheBoardPiece,
   setFellOffTheBoardPiece,
-  changeItemPlacedRecently,
-  wasItemPlacedRecently,
+  changeItemPlacedRecently: changewasItemPlacedThisTurn,
```
Comment (Ido-Barnea) [inline development/code/Game.ts]: fix please

File: development/code/ui/Events.ts [hunk 17]
```diff
@@ -87,11 +87,16 @@ export function dragElement(element: HTMLElement) {
     if (!initialElement) return;
 
     let parentContainer = initialElement.parentElement ?? undefined;
-    while (parentContainer && !(parentContainer.classList.contains('board') || parentContainer.classList.contains('player-inventory'))) {
+    while (
+      parentContainer &&
+       !(parentContainer.classList.contains('board') ||
+        parentContainer.classList.contains('player-inventory'))
```
Comment (Ido-Barnea) [inline development/code/ui/Events.ts:93]: extra space, why?

**Review by Ido-Barnea — CHANGES_REQUESTED** [review]

**Review by Ido-Barnea — APPROVED** [review]
ඞ ඞ ඞ ඞ ඞ ඞ ඞ
LGTM
ඞ ඞ ඞ ඞ ඞ ඞ ඞ
